src/js/events/VirtualNetworksActions.js

```diff
@@ -0,0 +1,55 @@
+import {RequestUtil} from 'mesosphere-shared-reactjs';
+
+import {
+  REQUEST_VIRTUAL_NETWORKS_SUCCESS,
+  REQUEST_VIRTUAL_NETWORKS_ERROR
+} from '../constants/ActionTypes';
+import AppDispatcher from './AppDispatcher';
+import Config from '../config/Config';
+
+const VirtualNetworksActions = {
+  fetch: function () {
+    RequestUtil.json({
+      url: `${Config.rootUrl}${Config.virtualNetworksApi}/state`,
+      success: function (response) {
+        let {overlays, vtep_mac_oui, vtep_subnet} = response.network;
+        // Map structure to mimic agents overlays
+        overlays = overlays.map(function (overlay) {
+          return {info: overlay};
+        });
```

Comment:
Yes, that is the reducer pattern that we introduced with ChronosActions: https://github.com/dcos/dcos-ui/blob/master/src/js/events/ChronosActions.js#L58